bazel/rules_rust.patch

```diff
@@ -9,3 +9,15 @@
          # These rules are not supposed to be depended on by other rust targets, and
          # as such they shouldn't provide a CrateInfo. However, one may still want to
          # write a rust_test for them, so we provide the CrateInfo wrapped in a provider
+
+--- rust/private/rustc.bzl
++++ rust/private/rustc.bzl
+@@ -1043,7 +1043,7 @@ def construct_arguments(
+ 
+     if toolchain.llvm_cov and ctx.configuration.coverage_enabled:
+         # https://doc.rust-lang.org/rustc/instrument-coverage.html
+-        rustc_flags.add("--codegen=instrument-coverage")
```

Comment:
Out of curiosity, why does this fix the wasm issue?

Reply:
We are compiling rust under `wasm-32-unkown-unkown` and `wasm-wasi` targets in our `//test` repo, but actually instrument-coverage/pgo is not supported for these target envs for rust. Since Envoy at current stage will never profile Rust code, we should be fine to disable it as a whole.